test/e2e/resize_nodes.go

```diff
@@ -0,0 +1,331 @@
+}
+
+func newService(name string) *api.Service {
+	return &api.Service{
+		ObjectMeta: api.ObjectMeta{
+			Name: "test-service",
+		},
+		Spec: api.ServiceSpec{
+			Selector: map[string]string{
+				"name": name,
+			},
+			Ports: []api.ServicePort{{
+				Port:       9376,
+				TargetPort: util.NewIntOrStringFromInt(9376),
+			}},
+		},
+	}
+}
+
+func createService(c *client.Client, ns, name string) {
+	_, err := c.Services(ns).Create(newService(name))
+	Expect(err).NotTo(HaveOccurred())
+}
+
+func newReplicationController(name string, replicas int, image string) *api.ReplicationController {
+	return &api.ReplicationController{
+		ObjectMeta: api.ObjectMeta{
+			Name: name,
+		},
+		Spec: api.ReplicationControllerSpec{
+			Replicas: replicas,
+			Selector: map[string]string{
+				"name": name,
+			},
+			Template: &api.PodTemplateSpec{
+				ObjectMeta: api.ObjectMeta{
+					Labels: map[string]string{"name": name},
+				},
+				Spec: api.PodSpec{
+					Containers: []api.Container{
+						{
+							Name:  name,
+							Image: image,
+							Ports: []api.ContainerPort{{ContainerPort: 9376}},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func createServeHostnameReplicationController(c *client.Client, ns, name string, replicas int) *api.ReplicationController {
+	By(fmt.Sprintf("creating replication controller %s", name))
+	rc, err := c.ReplicationControllers(ns).Create(newReplicationController(name, replicas, "gcr.io/google_containers/serve_hostname:1.1"))
+	Expect(err).NotTo(HaveOccurred())
+	return rc
+}
+
+func resizeReplicationController(c *client.Client, ns, name string, replicas int) {
+	rc, err := c.ReplicationControllers(ns).Get(name)
+	Expect(err).NotTo(HaveOccurred())
+	rc.Spec.Replicas = replicas
+	_, err = c.ReplicationControllers(rc.Namespace).Update(rc)
+	Expect(err).NotTo(HaveOccurred())
+}
+
+func waitForPodsCreated(c *client.Client, ns, name string, replicas int) *api.PodList {
+	// List the pods, making sure we observe all the replicas.
+	listTimeout := time.Minute
+	listInterval := 5 * time.Second
+	label := labels.SelectorFromSet(labels.Set(map[string]string{"name": name}))
+	var pods *api.PodList
+	t := time.Now()
+	for {
+		var err error
+		pods, err = c.Pods(ns).List(label, fields.Everything())
+		Expect(err).NotTo(HaveOccurred())
+
+		Logf("Controller %s: Found %d pods out of %d", name, len(pods.Items), replicas)
+		if len(pods.Items) == replicas {
+			break
+		}
+		if time.Since(t) > listTimeout {
+			Failf("Controller %s: Gave up waiting for %d pods to come up after seeing only %d pods after %v seconds",
+				name, replicas, len(pods.Items), time.Since(t).Seconds())
+		}
+		time.Sleep(listInterval)
+	}
+	return pods
+}
+
+func waitForPodsRunning(c *client.Client, pods *api.PodList) {
+	// Wait for the pods to enter the running state. Waiting loops until the pods
+	// are running so non-running pods cause a timeout for this test.
+	By("ensuring each pod is running")
+	for _, pod := range pods.Items {
```

Comment:
Because this `for` loop is sequential, this part of the test alone could take 5 minutes times the number of pods, when really we want it to only take 5 minutes total. Could you either parallelize this with channels, or, if that's too much work, add a TODO to do this?